services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/long_running_tasks_utils.py

```diff
@@ -40,3 +46,22 @@ async def run_before_shutdown_actions(
                     container_name,
                     exc_info=True,
                 )
+
+
+async def ensure_read_permissions_on_user_service_data(
+    mounted_volumes: MountedVolumes,
+) -> None:
+    # Makes sure sidecar has access to all files in the user services.
+    # The user could have removed read permissions form a file, which will cause an error.
+
+    # NOTE: command runs inside self container since the user service container might not always be running
+    self_container = os.environ["HOSTNAME"]
```

Comment:
Q: what is self_container? 
Q2: why you use os here instead of python `socket.gethostname()` ?

Reply:
Q: so you you revert the user permissions? What if the user has a legitimate reason to do that? i.e. when he recovers the data, the user will not get the same permissions he set?

Q2: is HOSTNAME always guaranteed?